docs: Flutter initialization examples and storage methods (#10233)
supabase/supabase: apps/docs/docs/reference/dart/initializing.mdx

final supabase = Supabase.instance.client;
```

### Call `initialize()` with custom headers

You can pass `headers` to initialize your Supabase client with customer headers. 
Here is an example of passing a custom auth header to Supabase client.
```dart title=main.dart
Future<void> main() async {
  await Supabase.initialize(
    url: 'https://xyzcompany.supabase.co',
    anonKey: 'public-anon-key',
    headers: {
      'Authorization': 'Bearer $accessToken',
    },
  );
  runApp(const MyApp());
}
```

### Call `initialize()`multiple times

> You can initialize Supabase multiple times in your app. When doing so, remember to call `dispose()` before every time you initialize again.

Reviewer: ```suggestion
You can initialize Supabase multiple times in your app. When doing so, call `dispose()` each time before you initialize again.
```
